python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+    namespace: Identifier = Field()
+    properties: Properties = Field()
+
+
+class UpdateNamespacePropertiesResponse(IcebergBaseModel):
+    removed: List[str] = Field()
+    updated: List[str] = Field()
+    missing: List[str] = Field()
+
+
+class ListTableResponseEntry(IcebergBaseModel):
+    name: str = Field()
+    namespace: Identifier = Field()
+
+
+class ListTablesResponse(IcebergBaseModel):
+    identifiers: List[ListTableResponseEntry] = Field()
+
+
+class ErrorResponseMessage(IcebergBaseModel):
+    message: str = Field()
+    type: str = Field()
+    code: int = Field()
+
+
+class ErrorResponse(IcebergBaseModel):
+    error: ErrorResponseMessage = Field()
+
+
+class RestCatalog(Catalog):
+    token: TokenResponse
+    config: ConfigResponse
+
+    host: str
+
+    def __init__(self, name: str, properties: Properties, host: str, token: str):
+        self.host = host
+        self.token = self._fetch_access_token(token)
+        self.config = self._fetch_config()
+        super().__init__(name, properties)
+
+    def _split_token(self, token: str) -> Tuple[str, str]:
+        """Splits the token in a client id and secret
+
+        Args:
+            token: The token with a semicolon as a separator
+
+        Returns:
+            The client id and secret
+        """
+        client, secret = token.split(":")
+        return client, secret
+
+    @property
+    def headers(self) -> Properties:
+        return {AUTHORIZATION_HEADER: f"{BEARER_PREFIX} {self.token.access_token}", "Content-type": "application/json"}
+
+    def url(self, endpoint: str, prefixed: bool = True, **kwargs) -> str:
+        """Constructs the endpoint
+
+        Args:
+            prefixed: If the prefix return by the config needs to be appended
+
+        Returns:
+            The base url of the rest catalog
+        """
+
+        url = self.host
+        url = url if url.endswith("/") else url + "/"
+
+        if prefixed:
+            url += self.config.overrides.get(PREFIX, "")
+            url = url if url.endswith("/") else url + "/"
+
+        return url + endpoint.format(**kwargs)
+
+    def _fetch_access_token(self, token: str) -> TokenResponse:
+        client, secret = self._split_token(token)
+        data = {GRANT_TYPE: CLIENT_CREDENTIALS, CLIENT_ID: client, CLIENT_SECRET: secret, SCOPE: CATALOG_SCOPE}
+        response = requests.post(self.url(Endpoints.get_token, prefixed=False), data=data)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {401: BadCredentialsError})
+
+        return TokenResponse(**response.json())
+
+    def _fetch_config(self) -> ConfigResponse:
+        response = requests.get(self.url(Endpoints.get_config, prefixed=False), headers=self.headers)
+        response.raise_for_status()
+        return ConfigResponse(**response.json())
+
+    def _split_namespace_and_table(self, identifier: Union[str, Identifier]) -> Properties:
```

Comment:
The difference between `_split_namespace_and_table` and `_split_namespace_and_name` is not very obvious, but it matters a lot which one you use! Is there a way to ensure that we are using the right one for JSON vs paths? Maybe we should have "path" in the name, like `_split_identifier_for_path` and `_split_identifier_for_json`?

Reply:
Yes, that's much better indeed